lib/src/main/kotlin/com/swmansion/starknet/data/serializers/JsonRpcErrorPolymorphicSerializer.kt

```diff
@@ -0,0 +1,60 @@
+package com.swmansion.starknet.data.serializers
+
+import com.swmansion.starknet.provider.rpc.JsonRpcContractError
+import com.swmansion.starknet.provider.rpc.JsonRpcError
+import com.swmansion.starknet.provider.rpc.JsonRpcStandardError
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+internal object JsonRpcErrorPolymorphicSerializer : JsonContentPolymorphicSerializer<JsonRpcError>(JsonRpcError::class) {
+    override fun selectDeserializer(element: JsonElement): DeserializationStrategy<JsonRpcError> {
+        val jsonElement = element.jsonObject
+
+        val isContractError = "data" in jsonElement &&
```

Comment:
Should base on code instead?

Reply:
if possible that would probably be a better approach

Reply:
Discussed offline, decided to leave the current approach to avoid unnecessary errors associated with invalid contract errors returned by nodes.